---
title: "Disparities"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
    <!-- margin-bottom: 100px !important; -->
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'#, fig.show='hold'
                      )
options(gtsummary.print_engine = "gt")
options(gtsummary.as_gt.addl_cmds = "gt::tab_options(table.font.size = 12, data_row.padding = gt::px(1))")

```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">Manuscript</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(data.table)
library(VennDiagram)
library(viridis)
library(lubridate)
library(gtsummary)
library(survival)
library(survminer)
```

```{r load}
# Global_data <- readRDS("/Users/colinccm/Documents/GitHub/CHIP-Avatar/Global_data.rds")
germline_patient_data <- readRDS("/Users/colinccm/Documents/GitHub/CHIP-Avatar/germline_patient_data.rds")
path <- ("/Users/colinccm/Documents/GitHub/Gillis/no vpn data/CHIP in Avatar")

germline_patient_data <- germline_patient_data %>% 
  rename(Sex = Gender) %>% 
  filter(is_patient_MM == "Yes" & is_MMDx_close_to_blood == "Yes") %>% 
  distinct(avatar_id, .keep_all = TRUE) %>% 
  filter(raceeth != "Others") %>% 
  mutate(raceeth = factor(raceeth, levels=c("White Non-Hispanic", "Hispanic", "Black"))) %>%
  mutate(ISS_at_MMdx = replace_na(ISS_at_MMdx, "Unknown")) %>%
  mutate(regimen_categoryVCD_1 = factor(regimen_categoryVCD_1, levels = c(
    "PI + IMIDs", "PI", "IMIDs", "Others"
  )))
```
<br>
<br>

I removed the "Others" in raceeth.
Missing values for ISS are imputed to the "unknown" category.


***

# Introduction

The ORIEN Avatar^TM Research Program focuses on patients with advanced primary or metastatic disease, those with limited treatment options, as well as patients who are likely to develop progressive disease, and will capitalize on the disease-area expertise of ORIEN's research scientists.   
It generates massive amounts of genetic and clinical information on patients consenting to the Total Cancer CareÂ® (TCC) Protocol.   
ORIEN Avatar tracks patient clinical progressions over their lifetime to provide the richest data source for targeted cancer therapy advancement.  ORIEN Avatar data solution is optimized for oncology research by providing integrated longitudinal clinical data and whole exome, transcriptome and germline sequencing around consented oncology patients.   
We are focusing on patients with multiple myeloma disease.
We will focus on patient with active MM who had WES analyzed on their blood and CH were analyzed.

<br>

***

# I. MM Avatar data description


```{r}
germline_patient_data <- germline_patient_data %>% 
  mutate(HCT_comparison = case_when(
    is.na(date_of_bmt_1)              ~ "No HCT",
    time_to_hct < 314                 ~ "Early HCT",
    time_to_hct >= 314                ~ "Late HCT"
  )) %>% 
  mutate(HCT_comparison = factor(HCT_comparison, levels = c("No HCT", "Early HCT", "Late HCT")))

contamination_tumor_ids <- paste("A025767", "A025771", "A007368", "A007322", "A000111", sep = "|")
```

<div class = "row">
<div class = "col-md-5">
```{r}
germ_data <- germline_patient_data %>% 
  mutate(Whole = "MM patients with WES") %>% 
  select(Age_at_MMonly_diagnosis, Sex, raceeth, Whole, ISS_at_MMdx) %>%
  tbl_summary(by = Whole, 
              label = list(raceeth ~ "Race/ethnicity", Age_at_MMonly_diagnosis ~ "Age at MM diagnosis"),
              sort = list(everything() ~ "frequency", ISS_at_MMdx ~ "alphanumeric"),
              digits = list(c(Age_at_MMonly_diagnosis) ~ 2)#,
              # missing = "no"
              ) %>% 
  bold_labels()
germ_data
```
</div>

<div class = "col-md-7">
```{r}
p <- qplot(x =Age_at_MMonly_diagnosis, data=subset(germline_patient_data,!is.na(Age_at_MMonly_diagnosis)), fill=..count.., geom="histogram") 
p + scale_fill_viridis_c(
  alpha = 1,
  begin = 0,
  end = 1,
  direction = 1,
  option = "D",
  values = NULL,
  space = "Lab",
  na.value = "grey50",
  guide = "colourbar",
  aesthetics = "fill"
) +
  theme_minimal(base_size = 16) +
  labs(x="Age at Diagnosis", y="Number of Patient", title="Repartition of Age at Diagnosis in MM Avatar")
```
</div>
</div>

<br>

Patients represented in MM Avatar data are characterized by a median age at diagnosis of `r inline_text(germ_data, variable = Age_at_MMonly_diagnosis, column = "MM patients with WES", pattern = "{median} years old")`. We saw slightly more male `r inline_text(germ_data, variable = Sex, level = "Male", column = "MM patients with WES")`, a majority of White Non-Hispanic `r inline_text(germ_data, variable = raceeth, level = "White Non-Hispanic", column = "MM patients with WES")`(Blacks `r inline_text(germ_data, variable = raceeth, level = "Black", column = "MM patients with WES")`, Hispanic `r inline_text(germ_data, variable = raceeth, level = "Hispanic", column = "MM patients with WES")`).
A higher proportion of patients have a lower clinical stage at blood with `r inline_text(germ_data, variable = ISS_at_MMdx, level = "I", column = "MM patients with WES")` patients with ISS I..

<br>
<br>
```{r}
germline_patient_data %>% 
  select(Disease_Status_germline) %>% 
  tbl_summary(
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels()
```
<br>
<br>


## Cytogenetics
```{r}
germline_patient_data %>% 
  select(fish_cytogenetics, cytogenetics_risk) %>% 
  tbl_summary(
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels()

germline_patient_data %>% 
  select(c("amp_dup_1q21", "del_1p", "del_17p", "del_13q",
              "t11_14", "t14_16", "t4_14")) %>% 
  tbl_summary(
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels()
```

<br>
<br>

***

# II. Age Distribution in Disparities

## Raceeth
```{r}
# tiff(paste0(path, "/Disparities manuscript/age distribution boxplot raceeth.tiff"))
p <- ggplot(germline_patient_data, 
            aes(x=raceeth, y=Age_at_MMonly_diagnosis)) + 
  geom_boxplot(color = c("#03051AFF", "blue", "red"),
               alpha = 0.3,
               fill = c("#03051AFF", "blue", "red")) + 
  theme_classic(base_size = 23) +
  labs(x=NULL, y="Age at Diagnosis", title="Age of MM Diagnosis Distribution")
p + geom_jitter(shape=16, position=position_jitter(0.2)) +
  stat_compare_means(label.x = 1.75, 
                     # aes(label = paste0("p = ", ..p.format..))
                   aes(
  label=ifelse(..p.format.. == 1e-04, "p<0.001", "Other")
  )
)
# dev.off()
```

```{r tbl for number}
raceeth_age <- germline_patient_data %>% 
  select(raceeth, Age_at_MMonly_diagnosis) %>%
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
```
<br>

Interestingly, we noticed a significant difference between disparities on the age at diagnosis. White Hispanics are diagnosed at a median age of `r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "White Non-Hispanic", pattern = "{median} years old")`. A significant earlier age at diagnosis is measured for Hispanics, `r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "Hispanic", pattern = "{median} years old")` , and for Blacks, `r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "Black", pattern = "{median} years old")`, compare to Whites Hispanics (`r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "p.value")`).

<br>
<br>

***

# III. RaceEthnicity

## Clinical

```{r}
tbl1 <-
  germline_patient_data %>% 
  mutate(ISS_at_MMdx = str_replace(ISS_at_MMdx, "Unknown", NA_character_)) %>% 
  select(Age_at_MMonly_diagnosis, Age_at_death, Sex, raceeth, ISS_at_MMdx,
         CH_status,
         HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_summary(by = raceeth, 
              label = list(Age_at_MMonly_diagnosis ~ "Age at MM diagnosis",
                           ISS_at_MMdx ~ "ISS at MM diagnosis",
                           HCT_ever ~ "HCT ever",
                           CH_status ~ "CH status"),
              sort = list(everything() ~ "frequency", ISS_at_MMdx ~ "alphanumeric"),
              digits = list(c(Age_at_MMonly_diagnosis) ~ 2)
              ) %>% 
  bold_labels() %>% add_p(test.args = regimen_categoryVCD_1 ~ list(workspace=2e9)) %>% bold_p(t = .05)
```


```{r}
tbl2 <- germline_patient_data %>% 
  select(raceeth,
         time_to_drug, time_to_hct) %>%
  tbl_summary(by = raceeth, 
              label = list(time_to_drug ~ "Time to drug (days)", time_to_hct ~ "Time to HCT (days)"),
              sort = list(everything() ~ "frequency"),
              missing = "no"
              ) %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)

tbl_stack(list(tbl1, tbl2))
```

```{r}
# germline_patient_data %>% 
#   select(time_to_drug, HCT_comparison, raceeth) %>%
#   tbl_summary(by = raceeth, 
#               label = list(time_to_drug ~ "Time to drug (days)", HCT_comparison ~ "Time to HCT (days)"),
#               sort = list(everything() ~ "frequency"),
#               missing = "no") %>% bold_labels() %>% add_overall() %>% add_p() %>% bold_p(t = .05)
```

<br>
<br>

## Survivals

<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
raceeth_surv <- germline_patient_data

mysurv <- Surv(time = raceeth_surv$month_at_os, event = raceeth_surv$os_event)
myplot <- survfit(mysurv~raceeth, data = raceeth_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
# pdf(paste0(path, "/Disparities manuscript/OS raceeth.pdf"))
myplot <- ggsurvplot(myplot, data = raceeth_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           legend.labs = c("White Non-Hispanic", "Hispanic", "Black"),
           palette = c("#03051AFF", "blue", "red"),
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
) + guides(colour = guide_legend(nrow = 3))
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

# dev.off()

tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_os, 
                             event = raceeth_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_os, 
             event = raceeth_surv$os_event) ~ raceeth + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + HCT_ever + regimen_categoryVCD_1, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
# eth_cox_os <- 
  tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
# eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
myplot <- survfit(mysurv~raceeth, data = raceeth_surv)
# pdf(paste0(path, "/Disparities manuscript/PFS raceeth.pdf"))

myplot <- ggsurvplot(myplot, data = raceeth_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           legend.labs = c("White Non-Hispanic", "Hispanic", "Black"),
           palette = c("#03051AFF", "blue", "red"),
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
) + guides(colour = guide_legend(nrow = 3))
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

# dev.off()

tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_progression_drug, 
                             event = raceeth_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_progression_drug, 
             event = raceeth_surv$drug_progression_event) ~ raceeth + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + strata(HCT_ever) + regimen_categoryVCD_1, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
# eth_cox_pfs <- 
  tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
# eth_cox_pfs
```
</div>
</div>

<br>
<br>

```{r}
germline_patient_data %>% 
  select(Disease_Status_germline, raceeth) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels() %>% add_p(test.args = Disease_Status_germline ~ list(workspace=2e9))
```
<br>
<br>

## Cytogenetics
```{r}
germline_patient_data %>% 
  select(fish_cytogenetics, cytogenetics_risk, raceeth) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)

germline_patient_data %>% 
  select(c("amp_dup_1q21", "del_1p", "del_17p", "del_13q",
              "t11_14", "t14_16", "t4_14", raceeth)) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
```


***

# IV. CH

From here, remove 5 patients with tumor contamination

```{r}
patients_clean_tumor <- germline_patient_data %>% 
  filter(!str_detect(avatar_id, contamination_tumor_ids))
```

```{r}
ch_char <- patients_clean_tumor %>% 
  mutate(Whole = "MM patients at time of blood with WES") %>% 
  select(Whole, CH_status) %>%
  tbl_summary(by = Whole, 
              label = list(CH_status ~ "CH status"),
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels()
ch_char
```
<br>
<br>

Whole Exome Sequencing data identified CH mutations in `r inline_text(ch_char, variable = CH_status, level = "CH", column = "MM patients at time of blood with WES")`.
+ Add bioinfo results (oncoplot?).  
Type of genes affected...

<br>
<br>

```{r}
# ch_samples <- patients_clean_tumor %>%
#   select(CH_status) %>% 
#   tbl_summary(sort = list(everything() ~ "frequency")) %>% 
#   bold_labels()
# ch_samples
patients_clean_tumor %>%
  mutate(ISS_at_MMdx = str_replace(ISS_at_MMdx, "Unknown", NA_character_)) %>% 
  select(Age_at_MMonly_diagnosis, 
         Sex, raceeth, ISS_at_MMdx, ISS_at_MMdx,
         CH_status) %>% 
  tbl_summary(by = CH_status, 
              label = list(Age_at_MMonly_diagnosis ~ "Age at MM diagnosis", CH_status ~ "CH status"),
              sort = list(everything() ~ "frequency", ISS_at_MMdx ~ "alphanumeric", ISS_at_MMdx ~ "alphanumeric"),
              digits = list(c(Age_at_MMonly_diagnosis) ~ 2),
              percent = "row") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)

ch_demo <- patients_clean_tumor %>%
  mutate(ISS_at_MMdx = str_replace(ISS_at_MMdx, "Unknown", NA_character_)) %>% 
  select(Age_at_MMonly_diagnosis, 
         Sex, raceeth, ISS_at_MMdx, ISS_at_MMdx,
         CH_status) %>% 
  tbl_summary(by = CH_status, 
              label = list(Age_at_MMonly_diagnosis ~ "Age at MM diagnosis", CH_status ~ "CH status"),
              sort = list(everything() ~ "frequency", ISS_at_MMdx ~ "alphanumeric", ISS_at_MMdx ~ "alphanumeric"),
              digits = list(c(Age_at_MMonly_diagnosis) ~ 2)) %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
ch_demo
```
<br>

A significant difference in the age at diagnosis was observed between CH and No CH patients with an older age for CH patients (`r inline_text(ch_demo, variable = Age_at_MMonly_diagnosis, column = "CH")` vs `r inline_text(ch_demo, variable = Age_at_MMonly_diagnosis, column = "No CH")`, `r inline_text(ch_demo, variable = Age_at_MMonly_diagnosis, column = "p.value")`). While Sex and race/ethnicity are similarly distributed (slightly more male), we noticed a higher proportion of patients with more severe clinical stage (II-III) for CH patients (`r inline_text(ch_demo, variable = ISS_at_MMdx, column = "p.value")`).

<br>
<br>

## Cytogenetics
```{r}
patients_clean_tumor %>% 
  select(fish_cytogenetics, cytogenetics_risk, CH_status) %>% 
  tbl_summary(by = CH_status, 
              sort = list(everything() ~ "frequency")) %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)

patients_clean_tumor %>% 
  select(c("amp_dup_1q21", "del_1p", "del_17p", "del_13q",
              "t11_14", "t14_16", "t4_14", CH_status)) %>% 
  tbl_summary(by = CH_status, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
```

***

# IV. CH & RaceEthnicity
### PFS
```{r, fig.height = 7}
raceeth_surv <- patients_clean_tumor

mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
# tiff(paste0(path, "/Disparities manuscript/PFS CH raceeth.tiff"))

myplot <- ggsurvplot(myplot, data = raceeth_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           color = "raceeth",
           linetype = "CH_status",
           palette = c("red", "blue", "#03051AFF"),
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
)+ guides(colour = guide_legend(nrow = 3), linetype = guide_legend(nrow = 2))
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

# dev.off()
```

```{r}
tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1, CH_status) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_progression_drug, 
                             event = raceeth_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_progression_drug, 
             event = raceeth_surv$drug_progression_event) ~ raceeth + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + strata(HCT_ever) + regimen_categoryVCD_1 + CH_status, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
```

### OS
```{r, fig.height = 7}
mysurv <- Surv(time = raceeth_surv$month_at_os, event = raceeth_surv$os_event)
myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
# tiff(paste0(path, "/Disparities manuscript/OS CH raceeth.tiff"))

myplot <- ggsurvplot(myplot, data = raceeth_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           color = "raceeth",
           linetype = "CH_status",
           palette = c("red", "blue", "#03051AFF"),
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
)+ guides(colour = guide_legend(nrow = 3), linetype = guide_legend(nrow = 2))
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

# dev.off()
```

```{r}
tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1, CH_status) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_os, 
                             event = raceeth_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_os, 
                   event = raceeth_surv$os_event) ~ raceeth + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + HCT_ever + regimen_categoryVCD_1 + CH_status, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
```
<br>
<br>


***
```{r, fig.height = 7}
# mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
# myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
# ggsurvplot(myplot, data = raceeth_surv, facet.by = "MM_age_cat5",
#            title = "PFS",
#            font.main = c(24, "bold", "black"),
#            font.x = c(20, "bold", "black"),
#            font.y = c(20, "bold", "black"),
#            font.legend = c(20, "bold", "black"),
#            font.tickslab = c(18, "bold", "black"),
#            size = 1.5,
# color = "raceeth",
#            linetype = "CH_status",
#            xlab = "Time in months"
# )+ guides(colour = guide_legend(nrow = 4))
```

```{r, fig.height = 7}
# mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
# myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
# ggsurvplot(myplot, data = raceeth_surv, facet.by = "MM_age_cat3",
#            title = "PFS",
#            font.main = c(24, "bold", "black"),
#            font.x = c(20, "bold", "black"),
#            font.y = c(20, "bold", "black"),
#            font.legend = c(20, "bold", "black"),
#            font.tickslab = c(18, "bold", "black"),
#            size = 1.5,
#            
#            xlab = "Time in months"
# )+ guides(colour = guide_legend(nrow = 4))
```
<br>
<br>

# Stratified by raceeth
## WNH
<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
nhis_surv <- patients_clean_tumor %>% 
  filter(raceeth == "White Non-Hispanic")

mysurv <- Surv(time = nhis_surv$month_at_os, event = nhis_surv$os_event)
myplot <- survfit(mysurv~CH_status, data = nhis_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
# tiff(paste0(path, "/Disparities manuscript/OS NHW.tiff"))

myplot <- ggsurvplot(myplot, data = nhis_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           linetype = "CH_status",
           color = "#03051AFF",
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
)
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

# dev.off()

tbl1 <- nhis_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = nhis_surv$month_at_os, 
                             event = nhis_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = nhis_surv$month_at_os, 
             event = nhis_surv$os_event) ~ CH_status + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + HCT_ever + regimen_categoryVCD_1, data =  nhis_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_os <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = nhis_surv$month_at_progression_drug, event = nhis_surv$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = nhis_surv)
myplot <- ggsurvplot(myplot, data = nhis_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           linetype = "CH_status",
           color = "#03051AFF",
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
)
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot


tbl1 <- nhis_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = nhis_surv$month_at_progression_drug, 
                             event = nhis_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = nhis_surv$month_at_progression_drug, 
             event = nhis_surv$drug_progression_event) ~ CH_status + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + strata(HCT_ever) + regimen_categoryVCD_1, data =  nhis_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_pfs <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_pfs
```
</div>
</div>

<br>
<br>

## Bl
<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
bl_surv <- patients_clean_tumor %>% 
  filter(raceeth == "Black")

mysurv <- Surv(time = bl_surv$month_at_os, event = bl_surv$os_event)
myplot <- survfit(mysurv~CH_status, data = bl_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
# tiff(paste0(path, "/Disparities manuscript/OS Bl.tiff"))

myplot <- ggsurvplot(myplot, data = bl_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           linetype = "CH_status",
           color = "red",
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),
 
           # Censor
           censor = TRUE
)
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

# dev.off()

tbl1 <- bl_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = bl_surv$month_at_os, 
                             event = bl_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = bl_surv$month_at_os, 
             event = bl_surv$os_event) ~ CH_status + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + HCT_ever + regimen_categoryVCD_1, data =  bl_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_os <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = bl_surv$month_at_progression_drug, event = bl_surv$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = bl_surv)
myplot <- ggsurvplot(myplot, data = bl_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           linetype = "CH_status",
           color = "red",
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
)
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot


tbl1 <- bl_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = bl_surv$month_at_progression_drug, 
                             event = bl_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = bl_surv$month_at_progression_drug, 
             event = bl_surv$drug_progression_event) ~ CH_status + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + strata(HCT_ever) + regimen_categoryVCD_1, data =  bl_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_pfs <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_pfs
```
</div>
</div>

<br>
<br>

## H

<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
his_surv <- patients_clean_tumor %>% 
  filter(raceeth == "Hispanic")

mysurv <- Surv(time = his_surv$month_at_os, event = his_surv$os_event)
myplot <- survfit(mysurv~CH_status, data = his_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
# tiff(paste0(path, "/Disparities manuscript/OS His.tiff"))

myplot <- ggsurvplot(myplot, data = his_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           linetype = "CH_status",
           color = "blue",
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
)
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot
# dev.off()

tbl1 <- his_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = his_surv$month_at_os, 
                             event = his_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = his_surv$month_at_os, 
             event = his_surv$os_event) ~ CH_status + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + HCT_ever + regimen_categoryVCD_1, data =  his_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_os <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = his_surv$month_at_progression_drug, event = his_surv$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = his_surv)
myplot <- ggsurvplot(myplot, data = his_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           linetype = "CH_status",
           color = "blue",
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400),

           # Censor
           censor = TRUE
)
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

tbl1 <- his_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Sex, ISS_at_MMdx, HCT_ever, regimen_categoryVCD_1) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = his_surv$month_at_progression_drug, 
                             event = his_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent(location = "level") %>% add_n(location = "level") %>%
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = his_surv$month_at_progression_drug, 
             event = his_surv$drug_progression_event) ~ CH_status + Age_at_MMonly_diagnosis + Sex + ISS_at_MMdx + strata(HCT_ever) + regimen_categoryVCD_1, data =  his_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_pfs <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_pfs
```
</div>
</div>

<br>

<!-- Age is significantly associated with OS `r inline_text(tbl1, variable = Age_at_MMonly_diagnosis, pattern = "(OR {estimate}; 95% CI {conf.low}, {conf.high}; {p.value})")`. -->

<br>

<!-- I thought it could be a year effect compare to Taiga said (maybe earlier it wasn't a difference like he said but now that Hispanic may have more access to care we can see the difference) but no. -->

# KM by decade
### Raceeth
```{r}
table(patients_clean_tumor$MM_age_cat)

mysurv <- Surv(time = patients_clean_tumor$month_at_os, event = patients_clean_tumor$os_event)
myplot <- survfit(mysurv~raceeth+MM_age_cat, data = patients_clean_tumor)
ggsurvplot_facet(myplot, data = patients_clean_tumor, facet.by = "MM_age_cat",
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE
)
```

```{r}
# mysurv <- Surv(time = patients_clean_tumor$month_at_progression_drug, event = patients_clean_tumor$drug_progression_event)
# myplot <- survfit(mysurv~raceeth+MM_age_cat, data = patients_clean_tumor)
# ggsurvplot_facet(myplot, data = patients_clean_tumor, facet.by = "MM_age_cat",
#            title = "PFS",
#            conf.int = FALSE
# )
```


```{r}
# mysurv <- Surv(time = patients_clean_tumor$month_at_os, event = patients_clean_tumor$os_event)
# myplot <- survfit(mysurv~raceeth+MM_age_cat3, data = patients_clean_tumor)
# ggsurvplot_facet(myplot, data = patients_clean_tumor, facet.by = "MM_age_cat3",
#            title = "OS",
#            short.panel.labs = TRUE,
#            conf.int = FALSE
# )
```

```{r}
# mysurv <- Surv(time = patients_clean_tumor$month_at_progression_drug, event = patients_clean_tumor$drug_progression_event)
# myplot <- survfit(mysurv~raceeth+MM_age_cat3, data = patients_clean_tumor)
# ggsurvplot_facet(myplot, data = patients_clean_tumor, facet.by = "MM_age_cat3",
#            title = "PFS",
#            conf.int = FALSE
# )
```

### CH

```{r}
mysurv <- Surv(time = patients_clean_tumor$month_at_os, event = patients_clean_tumor$os_event)
myplot <- survfit(mysurv~CH_status+MM_age_cat, data = patients_clean_tumor)
ggsurvplot_facet(myplot, data = patients_clean_tumor, facet.by = "MM_age_cat",
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE
)

# mysurv <- Surv(time = patients_clean_tumor$month_at_progression_drug, event = patients_clean_tumor$drug_progression_event)
# myplot <- survfit(mysurv~CH_status+MM_age_cat, data = patients_clean_tumor)
# ggsurvplot_facet(myplot, data = patients_clean_tumor, facet.by = "MM_age_cat",
#            title = "PFS",
#            conf.int = FALSE
# )
```

```{r}
temp <- patients_clean_tumor %>% filter(raceeth != "Hispanic")
mysurv <- Surv(time = temp$month_at_os, event = temp$os_event)

myplot <- survfit(mysurv~CH_status+raceeth, data = temp)
ggsurvplot_facet(myplot, data = temp, facet.by = "MM_age_cat",
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE
) + guides(colour = guide_legend(nrow = 4))
```

### <50
```{r, fig.height=7}
temp <- patients_clean_tumor %>% filter(raceeth != "Hispanic") %>% 
  filter(MM_age_cat == "<50")
mysurv <- Surv(time = temp$month_at_os, event = temp$os_event)

myplot <- survfit(mysurv~CH_status+raceeth, data = temp)
ggsurvplot(myplot, data = temp,
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE,
           risk.table = TRUE
) + guides(colour = guide_legend(nrow = 4))
```

### 50-59
```{r, fig.height=7}
temp <- patients_clean_tumor %>% filter(raceeth != "Hispanic") %>% 
  filter(MM_age_cat == "50-59")
mysurv <- Surv(time = temp$month_at_os, event = temp$os_event)

myplot <- survfit(mysurv~CH_status+raceeth, data = temp)
ggsurvplot(myplot, data = temp,
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE,
           risk.table = TRUE
) + guides(colour = guide_legend(nrow = 4))
```

### 60-69
```{r, fig.height=7}
temp <- patients_clean_tumor %>% filter(raceeth != "Hispanic") %>% 
  filter(MM_age_cat == "60-69")
mysurv <- Surv(time = temp$month_at_os, event = temp$os_event)

myplot <- survfit(mysurv~CH_status+raceeth, data = temp)
ggsurvplot(myplot, data = temp,
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE,
           risk.table = TRUE
) + guides(colour = guide_legend(nrow = 4))
```

### >70
```{r, fig.height=7}
temp <- patients_clean_tumor %>% filter(raceeth != "Hispanic") %>% 
  filter(MM_age_cat == "â¥70")
mysurv <- Surv(time = temp$month_at_os, event = temp$os_event)

myplot <- survfit(mysurv~CH_status+raceeth, data = temp)
ggsurvplot(myplot, data = temp,
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE,
           risk.table = TRUE
) + guides(colour = guide_legend(nrow = 4))
```

### Per group
### <60
```{r, fig.height=7}
temp <- patients_clean_tumor %>% filter(raceeth != "Hispanic") %>% 
  filter(MM_age_group == "<60")
mysurv <- Surv(time = temp$month_at_os, event = temp$os_event)

myplot <- survfit(mysurv~CH_status+raceeth, data = temp)
ggsurvplot(myplot, data = temp,
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE,
           risk.table = TRUE
) + guides(colour = guide_legend(nrow = 4))
```

### â¥60
```{r, fig.height=7}
temp <- patients_clean_tumor %>% filter(raceeth != "Hispanic") %>% 
  filter(MM_age_group == "â¥60")
mysurv <- Surv(time = temp$month_at_os, event = temp$os_event)

myplot <- survfit(mysurv~CH_status+raceeth, data = temp)
ggsurvplot(myplot, data = temp,
           title = "OS",
           short.panel.labs = TRUE,
           conf.int = FALSE,
           risk.table = TRUE
) + guides(colour = guide_legend(nrow = 4))
```

# CH genes
```{r}
path <- fs::path("","Volumes","Gillis_Research","Christelle Colin-Leitzinger", "CHIP in Avatar")

ch_genes <- read_csv(paste0(path, "/Nancy's working files/CH calls_WES_genes_12.10.20.csv")) %>% 
  distinct(patient_id, GENE) %>% 
  mutate(patient_id = str_remove(patient_id, "_normal")) %>% 
  mutate(ARID1B = ifelse(GENE == "ARID1B", "Yes", NA_character_)) %>% 
  mutate(ASXL1 = ifelse(GENE == "ASXL1", "Yes", NA_character_)) %>% 
  mutate(ATM = ifelse(GENE == "ATM", "Yes", NA_character_)) %>% 
  mutate(CBL = ifelse(GENE == "CBL", "Yes", NA_character_)) %>% 
  mutate(CHEK2 = ifelse(GENE == "CHEK2", "Yes", NA_character_)) %>% 
  mutate(DNMT3A = ifelse(GENE == "DNMT3A", "Yes", NA_character_)) %>% 
  mutate(FLT3 = ifelse(GENE == "FLT3", "Yes", NA_character_)) %>% 
  mutate(IDH1 = ifelse(GENE == "IDH1", "Yes", NA_character_)) %>%
  mutate(JAK1 = ifelse(GENE == "JAK1", "Yes", NA_character_)) %>%
  mutate(JAK2 = ifelse(GENE == "JAK2", "Yes", NA_character_)) %>%
  mutate(KMT2D = ifelse(GENE == "KMT2D", "Yes", NA_character_)) %>%
  mutate(NOTCH1 = ifelse(GENE == "NOTCH1", "Yes", NA_character_)) %>%
  mutate(PPM1D = ifelse(GENE == "PPM1D", "Yes", NA_character_)) %>%
  mutate(RUNX1 = ifelse(GENE == "RUNX1", "Yes", NA_character_)) %>%
  mutate(SF3B1 = ifelse(GENE == "SF3B1", "Yes", NA_character_)) %>%
  mutate(STAG2 = ifelse(GENE == "STAG2", "Yes", NA_character_)) %>%
  mutate(STAT5B = ifelse(GENE == "STAT5B", "Yes", NA_character_)) %>%
  mutate(TET2 = ifelse(GENE == "TET2", "Yes", NA_character_)) %>%
  mutate(TP53 = ifelse(GENE == "TP53", "Yes", NA_character_)) %>%
  mutate(ZRSR2 = ifelse(GENE == "ZRSR2", "Yes", NA_character_)) %>%
  select(patient_id, ARID1B : ZRSR2) %>% 
  group_by(patient_id) %>% 
  fill(everything(), .direction = "updown") %>% 
  ungroup() %>% 
  distinct(patient_id, .keep_all = TRUE) %>% 
  mutate(across(where(is.character), .fns = ~ replace_na(., "No"))) %>% 
  right_join(., patients_clean_tumor, by = c("patient_id" = "SLID_germline"))
  
ch_genes %>% 
  select("ASXL1", "DNMT3A", "TET2", "TP53", "PPM1D",
         "SF3B1", "CBL", "JAK1", "STAT5B", 
         "ATM", "CHEK2",
         "FLT3", 
         "JAK2", "KMT2D", "NOTCH1", "RUNX1",
         "ZRSR2",
         "ARID1B", "IDH1", "STAG2"
         ) %>% 
  tbl_summary(type = list(all_categorical() ~ "dichotomous"),
              missing = "no") %>% 
  as_gt() %>% 
  gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

ch_genes %>% 
  select("ASXL1", "DNMT3A", "TET2", "TP53", "PPM1D",
         "SF3B1", "CBL", "JAK1", "STAT5B", 
         "ATM", "CHEK2",
         "FLT3", 
         "JAK2", "KMT2D", "NOTCH1", "RUNX1",
         "ZRSR2",
         "ARID1B", "IDH1", "STAG2", raceeth) %>% 
  tbl_summary(by = raceeth,
              type = list(all_categorical() ~ "dichotomous"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05) %>%
  as_gt() %>% 
  gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

```


# Tumor mutations

```{r}
library(maftools)
path <- fs::path("","Volumes","Gillis_Research","Christelle Colin-Leitzinger", "merging slid ID")

tumor <- read_csv(paste0(path, "/List tumor SLID earliest or closest to germline.csv")) %>%
  select(avatar_id, SLID_tumor) %>%
  rename(Tumor_Sample_Barcode = SLID_tumor) %>% 
  right_join(., patients_clean_tumor, by = "avatar_id")

# slids <- read_csv(paste0(path, "/list RNA SLID corresponding to DNA SLID 03222021.csv")) %>% 
#   select("avatar_id", "SLID_germline", "SLID_tumor", "rna_slid")
# 
# tumor1 <- right_join(tumor, slids, by = c("avatar_id", "Tumor_Sample_Barcode" = "SLID_tumor"))
# slids_list <- right_join(tumor1, patients_clean_tumor, by = c("avatar_id", "SLID_germline")) %>% 
#   select("avatar_id", "DNASequencingLibraryID" = "Tumor_Sample_Barcode", "RNASequencingLibraryID" = "rna_slid",
#          "SLID_germline", raceeth)


path <- fs::path("","Volumes","Gillis_Research","Christelle Colin-Leitzinger", "CHIP in Avatar")

gene_list <- c("KRAS", "NRAS", "FAM46C", "DIS3", "BRAF", "TP53", "RYR1", "DNAH5", "LRP1B",
               "TRAF3", "EGR1", "SP140", "PRKD2", "CYLD", "RB1", "IRF4", "CSMD3", 
               "AUTS2", "ABI3BP", "GRM7", "PARP4", "BCL7A", "SPEF2", "MYH13", "BRWD3", 
               "MAX", "RPL10", "DDX17", "SAMHD1", "PLD1", "ANKRD26", "ATM", "CCND1", "SETD2")

tumor_mutation_avatar <- read_csv(paste0(path, "/TumorMuts/processed data/somatic mutation from Jamie MAF file.csv")) %>% 
  filter(Hugo_Symbol %in% gene_list) %>% 
  right_join(., tumor, by = "avatar_id")

# tumor_mutation_avatar %>% 
#   distinct(avatar_id, Hugo_Symbol, .keep_all = TRUE) %>% 
#   select(Hugo_Symbol) %>% 
#   tbl_summary(sort = list(everything() ~ "frequency")) %>% 
#   as_gt() %>% 
#   gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

df <- tumor_mutation_avatar %>% 
  distinct(avatar_id, Hugo_Symbol, .keep_all = TRUE) %>% 
  mutate(KRAS = ifelse(Hugo_Symbol == "KRAS", "Yes", NA_character_)) %>% 
  mutate(NRAS = ifelse(Hugo_Symbol == "NRAS", "Yes", NA_character_)) %>% 
  mutate(FAM46C = ifelse(Hugo_Symbol == "FAM46C", "Yes", NA_character_)) %>% 
  mutate(DIS3 = ifelse(Hugo_Symbol == "DIS3", "Yes", NA_character_)) %>% 
  mutate(BRAF = ifelse(Hugo_Symbol == "BRAF", "Yes", NA_character_)) %>% 
  mutate(TP53 = ifelse(Hugo_Symbol == "TP53", "Yes", NA_character_)) %>% 
  mutate(RYR1 = ifelse(Hugo_Symbol == "RYR1", "Yes", NA_character_)) %>% 
  mutate(DNAH5 = ifelse(Hugo_Symbol == "DNAH5", "Yes", NA_character_)) %>% 
  mutate(LRP1B = ifelse(Hugo_Symbol == "LRP1B", "Yes", NA_character_)) %>% 
  mutate(TRAF3 = ifelse(Hugo_Symbol == "TRAF3", "Yes", NA_character_)) %>% 
  mutate(EGR1 = ifelse(Hugo_Symbol == "EGR1", "Yes", NA_character_)) %>% 
  mutate(SP140 = ifelse(Hugo_Symbol == "SP140", "Yes", NA_character_)) %>% 
  mutate(PRKD2 = ifelse(Hugo_Symbol == "PRKD2", "Yes", NA_character_)) %>% 
  mutate(CYLD = ifelse(Hugo_Symbol == "CYLD", "Yes", NA_character_)) %>% 
  mutate(RB1 = ifelse(Hugo_Symbol == "RB1", "Yes", NA_character_)) %>% 
  mutate(IRF4 = ifelse(Hugo_Symbol == "IRF4", "Yes", NA_character_)) %>% 
  mutate(CSMD3 = ifelse(Hugo_Symbol == "CSMD3", "Yes", NA_character_)) %>% 
  mutate(AUTS2 = ifelse(Hugo_Symbol == "AUTS2", "Yes", NA_character_)) %>% 
  mutate(ABI3BP = ifelse(Hugo_Symbol == "ABI3BP", "Yes", NA_character_)) %>% 
  mutate(GRM7 = ifelse(Hugo_Symbol == "GRM7", "Yes", NA_character_)) %>% 
  mutate(PARP4 = ifelse(Hugo_Symbol == "PARP4", "Yes", NA_character_)) %>% 
  mutate(BCL7A = ifelse(Hugo_Symbol == "BCL7A", "Yes", NA_character_)) %>% 
  mutate(SPEF2 = ifelse(Hugo_Symbol == "SPEF2", "Yes", NA_character_)) %>%
  mutate(MYH13 = ifelse(Hugo_Symbol == "MYH13", "Yes", NA_character_)) %>% 
  mutate(BRWD3 = ifelse(Hugo_Symbol == "BRWD3", "Yes", NA_character_)) %>%
  mutate(MAX = ifelse(Hugo_Symbol == "MAX", "Yes", NA_character_)) %>% 
  mutate(RPL10 = ifelse(Hugo_Symbol == "RPL10", "Yes", NA_character_)) %>% 
  mutate(DDX17 = ifelse(Hugo_Symbol == "DDX17", "Yes", NA_character_)) %>%
  mutate(SAMHD1 = ifelse(Hugo_Symbol == "SAMHD1", "Yes", NA_character_)) %>%
  mutate(PLD1 = ifelse(Hugo_Symbol == "PLD1", "Yes", NA_character_)) %>% 
  mutate(ANKRD26 = ifelse(Hugo_Symbol == "ANKRD26", "Yes", NA_character_)) %>% 
  mutate(ATM = ifelse(Hugo_Symbol == "ATM", "Yes", NA_character_)) %>% 
  mutate(CCND1 = ifelse(Hugo_Symbol == "CCND1", "Yes", NA_character_)) %>% 
  mutate(SETD2 = ifelse(Hugo_Symbol == "SETD2", "Yes", NA_character_)) %>% 
  
  mutate(BRAF_V600 = case_when(
    Hugo_Symbol == "BRAF" &
      str_detect(HGVSp_Short, "p.V600")       ~ "Yes",
    TRUE                                      ~ NA_character_
  )) %>% 
  
  select("KRAS", "NRAS", "TP53", "DIS3", "BRAF", "LRP1B", "FAM46C",
         "TRAF3", "CSMD3", "SP140", "CYLD", "DNAH5", "BRAF V600" = BRAF_V600, "RYR1", 
         "ATM", "MAX", 
         "IRF4", "BCL7A", "CCND1", "EGR1", "AUTS2", "PRKD2", "MYH13", "RB1",
         "PARP4", "SETD2", 
         "ANKRD26", "RPL10", "SAMHD1", "ABI3BP", "BRWD3",
         "GRM7", "PLD1",
         "SPEF2", "DDX17", raceeth, avatar_id) %>% 
  group_by(avatar_id) %>% 
  fill(everything(), .direction = "updown") %>% 
  ungroup() %>% 
  distinct(avatar_id, .keep_all = TRUE) %>% 
  select(-avatar_id) %>% 
  mutate(across(where(is.character), .fns = ~ replace_na(., "No"))) 

# tbl <- 
  df %>% 
  select(-raceeth) %>% 
  tbl_summary(sort = list(all_dichotomous() ~ "frequency")) %>% 
  as_gt() %>% 
  gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

# gt::gtsave(tbl, zoom = 1, paste0(path, "/TumorMuts/Output/Tables/Tumor Mutations in MM Avatar.pdf"))

# tumor_mutation_avatar %>% 
#   distinct(avatar_id, Hugo_Symbol, .keep_all = TRUE) %>% 
#   select(Hugo_Symbol, raceeth) %>% 
#   tbl_summary(by = raceeth,
#               sort = list(everything() ~ "frequency"),
#               missing = "no") %>% 
#   bold_labels() %>% add_p() %>% 
#   as_gt() %>% 
#   gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

# tbl <- 
  df %>% 
  tbl_summary(by = raceeth,
              sort = list(everything() ~ "frequency"),
              type = list(all_categorical() ~ "dichotomous"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05) %>%
  as_gt() %>% 
  gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

gene_list2 <- c("KRAS", "NRAS", "TP53", "DIS3", "BRAF",
               "SP140", "IRF4", "BCL7A", "AUTS2", "PARP4", "SETD2")

df %>% 
  select(gene_list2, raceeth) %>% 
  tbl_summary(by = raceeth,
              sort = list(everything() ~ "frequency"),
              type = list(all_categorical() ~ "dichotomous"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05) %>% add_q() %>% 
  as_gt() %>% 
  gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

df %>% 
  select(gene_list2, raceeth) %>%
  mutate(across(everything(), na_if, "No")) %>% 
  pivot_longer(cols = KRAS:SETD2) %>% 
  filter(!is.na(value)) %>% 
  
  group_by(raceeth,name) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100)
  ) %>% 
  mutate(name = factor(name, levels = c("KRAS", "NRAS", "TP53", "DIS3", "BRAF",
               "SP140", "IRF4", "BCL7A", "AUTS2", "PARP4", "SETD2"))) %>% 
  ggplot(aes(x=raceeth, y= percent, fill=name))+
  geom_bar(stat="identity") +
  scale_fill_viridis_d() +
  labs(fill = "Genes \nmutated", x = NULL, y = "Prevalence of Genes Mutations (percent)")+
  theme_classic()

df %>% 
  select(gene_list2, raceeth) %>%
  mutate(across(everything(), na_if, "No")) %>% 
  pivot_longer(cols = KRAS:SETD2) %>% 
  filter(!is.na(value)) %>% 
  
  group_by(raceeth,name) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100)
  ) %>% 
  mutate(name = factor(name, levels = c("KRAS", "NRAS", "TP53", "DIS3", "BRAF",
               "SP140", "IRF4", "BCL7A", "AUTS2", "PARP4", "SETD2"))) %>% 
  mutate(ymax = cumsum(percent)) %>% 
  mutate(ymin = c(0, head(ymax, n=-1))) %>% 
  
  mutate(labelPosition = (ymax + ymin) / 2) %>% 
  mutate(label = paste0(name, "\n value: ", round(percent, 0))) %>% 

  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=name)) +
     geom_rect() +
  
     geom_label(x=4, aes(y=labelPosition, label=label), size=3) +
  
     coord_polar(theta="y") + # Try to remove that to understand how the chart is built initially
     xlim(c(2, 4))+
  theme_void() +
  theme(legend.position = "none")+
  facet_wrap(.~ raceeth)
  

library(ggradar)
mycols <- c("#03051AFF", "blue", "red")
df %>% 
  select(gene_list2, raceeth) %>%
  mutate(across(everything(), na_if, "No")) %>% 
  pivot_longer(cols = KRAS:SETD2) %>% 
  filter(!is.na(value)) %>% 
  
  group_by(raceeth,name) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100)
  ) %>% 
  select(-count) %>% 
  mutate(name = factor(name, levels = c("KRAS", "NRAS", "TP53", "DIS3", "BRAF",
               "SP140", "IRF4", "BCL7A", "AUTS2", "PARP4", "SETD2"))) %>% 
  pivot_wider(names_from = name, values_from = percent) %>% 
  mutate(across(everything(), ~ replace_na(., 0)))  %>%
  ggradar(
    grid.min = 0, grid.mid = 15,
          grid.max = 30,
    values.radar = c("0", "15%", "30%"),
    group.point.size = 3,
    label.gridline.min = FALSE, 
    label.gridline.mid = FALSE,
    label.gridline.max = FALSE,
    # gridline.label.offset = 7 ,
    group.colours = mycols,
    axis.line.colour = "gray",
        gridline.min.colour = "gray",
        gridline.mid.colour = "gray",
        gridline.max.colour = "gray",
    # font.radar = "roboto",
    # grid.label.size = 13,  # Affects the grid annotations (0%, 50%, etc.)
    # axis.label.size = 8.5, # Afftects the names of the variables
    # group.point.size = 3   # Simply the size of the point 
    background.circle.colour = "white",
        # gridline.min.linetype = 1,
        # gridline.mid.linetype = 1,
        # gridline.max.linetype = 1,
    legend.position = "bottom"
  )

gene_list2 <- c("KRAS", "NRAS", "TP53", "DIS3", "BRAF",
               "SP140", "IRF4", "BCL7A", "AUTS2", "PARP4", "SETD2")










df %>% 
  select(gene_list2, raceeth) %>% 
  mutate(raceeth = factor(raceeth, levels = c("White Non-Hispanic", "Hispanic"))) %>% 
  tbl_summary(by = raceeth,
              sort = list(everything() ~ "frequency"),
              type = list(all_categorical() ~ "dichotomous"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05) %>% add_q() %>% 
  as_gt() %>% 
  gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

df %>% 
  select(gene_list2, raceeth) %>% 
  mutate(raceeth = factor(raceeth, levels = c("White Non-Hispanic", "Black"))) %>% 
  tbl_summary(by = raceeth,
              sort = list(everything() ~ "frequency"),
              type = list(all_categorical() ~ "dichotomous"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05) %>% add_q() %>% 
  as_gt() %>% 
  gt::tab_source_note(gt::md("**Each gene is counted once per patient**"))

# gt::gtsave(tbl, zoom = 1, paste0(path, "/TumorMuts/Output/Tables/Tumor Mutations in MM Avatar by race.pdf"))
```


```{r read maf, include=FALSE}
# maf1 <- 
#   read.maf(maf = paste0(path, "/TumorMuts/raw data/Nancy650AF.maf"), 
#            vc_nonSyn=c("frameshift_deletion", "frameshift_insertion",
#                        "nonframeshift_deletion", "nonframeshift_insertion",
#                        "splicing", "stopgain_SNV", "stoploss_SNV", "nonsynonymous_SNV"))
# 
# oncoplot(maf1, genes = gene_list, colors = col, sortByAnnotation = TRUE)

maf1 <- 
  read.maf(maf = paste0(path, "/TumorMuts/raw data/Nancy650AF.maf"), 
           vc_nonSyn=c("frameshift_deletion", "frameshift_insertion",
                       "nonframeshift_deletion", "nonframeshift_insertion",
                       "splicing", "stopgain_SNV", "stoploss_SNV", "nonsynonymous_SNV"), 
           clinicalData = tumor_mutation_avatar)

# maf_file <- read_delim(paste0(path, "/TumorMuts/raw data/Nancy650AF.maf"),
#                        delim = "\t") %>% 
#   filter(Variant_Classification == "frameshift_deletion"|
#          Variant_Classification == "frameshift_insertion"|
#          Variant_Classification == "nonframeshift_deletion"|
#          Variant_Classification == "nonframeshift_insertion"|
#          Variant_Classification == "splicing"|
#          Variant_Classification == "stopgain_SNV"|
#          Variant_Classification == "stoploss_SNV"|
#          Variant_Classification == "nonsynonymous_SNV"
#          ) %>%
#   mutate(Variant_Classification = str_replace(Variant_Classification, "_", " "),
#          Variant_Classification = str_to_sentence(Variant_Classification), 
#          Variant_Classification = str_replace(Variant_Classification, "snv", "SNV"))
#   
# 
# maf2 <- read.maf(maf = maf_file, 
#          vc_nonSyn=c("Frameshift deletion", "Frameshift insertion",
#                      "Nonframeshift deletion", "Nonframeshift insertion",
#                      "Splicing", "Stopgain SNV", "Stoploss SNV", "Nonsynonymous SNV"), 
#          clinicalData = tumor_mutation_avatar)
```


```{r oncoplot}
disparities_maf <- subsetMaf(maf = maf1, tsb = c(tumor_mutation_avatar$Tumor_Sample_Barcode))
# disparities_maf2 <- subsetMaf(maf = maf2, tsb = c(tumor_mutation_avatar$Tumor_Sample_Barcode))

fabcolors = c("#03051AFF", "blue", "red")
names(fabcolors) = c("White Non-Hispanic", "Hispanic", "Black")
fabcolors = list(raceeth = fabcolors)
# print(fabcolors)

col = c("yellow", "chartreuse3", "black", "steelblue2", "darkviolet", "cyan", "blue", "darkblue", "red"
        )
names(col) = c("frameshift_deletion", "frameshift_insertion",
                       "nonframeshift_deletion", "nonframeshift_insertion",
                       "splicing", "stopgain_SNV", "stoploss_SNV", "nonsynonymous_SNV", "Multi_Hit")

# col1 = c("yellow", "chartreuse3", "black", "steelblue2", "darkviolet", "cyan", "blue", "darkblue", "red"
#         )
# names(col1) = c("Frameshift deletion", "Frameshift insertion",
#                      "Nonframeshift deletion", "Nonframeshift insertion",
#                      "Splicing", "Stopgain SNV", "Stoploss SNV", "Nonsynonymous SNV", "Multi_Hit")
# tiff(paste0(path, "/TumorMuts/Output/Figures/oncoplot by raceeth.tiff"))
# tiff(paste0(path, "/Disparities manuscript/oncoplot by raceeth.tiff"))

# pdf(paste0(path, "/TumorMuts/Output/Figures/oncoplot by raceeth.pdf"), height = 6, width = 9)

oncoplot(disparities_maf, genes = gene_list2, colors = col,  clinicalFeatures = "raceeth", sortByAnnotation = TRUE, 
         annotationColor = fabcolors, annotationOrder = c("White Non-Hispanic", "Hispanic", "Black"),
         anno_height = .5, legend_height = 3, groupAnnotationBySize = FALSE
         )
# text(0.3, 0.5, "White Non-Hispnic", cex = 2, col = "red")
# oncoplot(disparities_maf2, genes = gene_list2, colors = col1,  clinicalFeatures = "raceeth", sortByAnnotation = TRUE, 
#          annotationColor = fabcolors, annotationOrder = c("White Non-Hispanic", "Hispanic", "Black"),
#          anno_height = .5, legend_height = 3, groupAnnotationBySize = FALSE
#          )
# dev.off()
```

```{r}
mmgenes <- readRDS("/Users/colinccm/Documents/GitHub/CHIP-Avatar/mmgenes.rds")

mmgenes %>% 
    ggplot(aes(x = raceeth, y = expression, fill = raceeth, color = raceeth))+
    geom_boxplot(
      # color = c("#03051AFF", "blue", "red"),
                 alpha = 0.3,
                 # fill = c("#03051AFF", "blue", "red")
      ) +
    scale_color_manual(values = c("#03051AFF", "blue", "red"))+
    scale_fill_manual(values = c("#03051AFF", "blue", "red"))+
    theme_classic()+
    scale_x_discrete(labels=c("White Non-Hispanic" = "White \nNon-Hispanic"))+
    theme(legend.position = "none", axis.title.x = element_blank())+
    facet_grid(.~ gene)

mmgenes %>% 
  group_by(raceeth, gene) %>% 
  summarize(mean = mean(expression),
            sd = sd(expression)) %>%
  mutate(fold = case_when(
    raceeth == "White Non-Hispanic"  ~ 1,
    gene == "BCL7A"                  ~ (mean / 3.95)-1,
    gene == "SPEF2"                  ~ (mean / 6.46)-1,
    gene == "ANKRD26"                ~ (mean / 7.10)-1
  )) %>% 
  mutate(sdfold = case_when(
    raceeth == "White Non-Hispanic"  ~ 0,
    gene == "BCL7A"                  ~ (sd / 2.14)-1,
    gene == "SPEF2"                  ~ (sd / 1.60)-1,
    gene == "ANKRD26"                ~ (sd / 0.821)-1
  )) %>% 
  filter(raceeth != "White Non-Hispanic") %>% 
  ggplot(aes(x=raceeth, y= fold, fill=raceeth))+
  geom_bar(stat="identity") +
  geom_errorbar(aes(x = raceeth,
                    ymin = fold - sdfold,
                    ymax = fold + sdfold),
                width = 0.2, alpha = 0.5,
                position=position_dodge(width=0.9))+
  scale_fill_viridis_d() +
  labs(fill = "Genes \nmutated", x = NULL, y = "Prevalence of Genes Mutations (percent)")+
  theme_classic()+
    facet_grid(.~ gene)
  
# mmgenes %>% 
#   group_by(raceeth, gene) %>% 
#   summarize(mean = mean(expression),
#             sd = sd(expression)) %>% 
#     ggplot(aes(x = raceeth, y = mean, fill = raceeth, color = raceeth))+
#     geom_point(
#       ) +
#   geom_errorbar(aes(x = raceeth,
#                     ymin = mean - sd,
#                     ymax = mean + sd),
#                 width = 0.2, alpha = 0.5,
#                 position=position_dodge(width=0.9))+
#     scale_color_manual(values = c("#03051AFF", "blue", "red"))+
#     scale_fill_manual(values = c("#03051AFF", "blue", "red"))+
#     theme_classic()+
#     scale_x_discrete(labels=c("White Non-Hispanic" = "White \nNon-Hispanic"))+
#     theme(legend.position = "none", axis.title.x = element_blank())+
#     facet_grid(.~ gene)

```

<!-- ## BRAF V600 -->
```{r}
# tumor_mutation_avatar %>% 
#   filter(Hugo_Symbol == "BRAF") %>% 
#   filter(str_detect(HGVSp_Short, "p.V600")) %>% 
#   distinct(avatar_id, .keep_all = TRUE) %>% 
#   select(Hugo_Symbol, raceeth) %>% 
#   tbl_summary(by = raceeth,
#               sort = list(everything() ~ "frequency"),
#               # type = list(all_categorical() ~ "dichotomous"),
#               missing = "no") %>% 
#   bold_labels() %>% add_p() %>% bold_p(t = .05)
```


***

# Others

## Drugs (only regimen1)

Ken that sometimes VCD needs to be put apart but with real explanation as when.
VCD apart
```{r}
patients_clean_tumor %>% 
  mutate(Whole = "MM patients with WES") %>% 
  select(regimen_category_1, raceeth) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p(test.args = regimen_category_1 ~ list(workspace=2e9)) %>% bold_p(t = .05)
```

VCD within PI as Taiga advise but still don't know what option to choose
```{r}
patients_clean_tumor %>% 
  mutate(Whole = "MM patients with WES") %>% 
  select(regimen_categoryVCD_1, raceeth) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p(test.args = regimen_categoryVCD_1 ~ list(workspace=2e9)) %>% bold_p(t = .05)

# fisher.test(tbl1$raceeth, tbl1$regimen_category_1)
# # fisher.test(tbl1$raceeth, tbl1$regimen_category_1, simulate.p.value = TRUE, B = 1e5)
# fisher.test(tbl1$raceeth, tbl1$regimen_category_1, workspace=2e9)
# fisher.test(tbl1$raceeth, tbl1$regimen_category_1, workspace=2e9, control='mult')
```
<br>

More than 20 patients
```{r}
patients_clean_tumor %>% 
  group_by(first_regimen_name) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20) %>% 
  # select(first_regimen_name, n)
  mutate(Whole = "MM patients with WES") %>% 
  select(first_regimen_name, raceeth) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
```
<br>

***

## CH

### Survivals

<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = patients_clean_tumor$month_at_os, event = patients_clean_tumor$os_event)
myplot <- survfit(mysurv~CH_status, data = patients_clean_tumor)
# jpeg(paste0(path, "/Figures/Survivals/CHIP/OS by CH.jpeg"), width = 1200, height = 900)
myplot <- ggsurvplot(myplot, data = patients_clean_tumor,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "", 
           linetype = "CH_status",
           color = "steelblue3",
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400)
)
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = patients_clean_tumor$month_at_progression_drug, event = patients_clean_tumor$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = patients_clean_tumor)
myplot <- ggsurvplot(myplot, data = patients_clean_tumor,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "", 
           linetype = "CH_status",
           color = "steelblue3",
           # # legend.labs = c("CH", "No CH"),
           pval = TRUE,
           pval.coord = c(0, 0.1),
           conf.int = FALSE,
           xlim = c(0, 400)
) 
myplot$plot <- myplot$plot + 
  scale_x_continuous(breaks = c(seq(0, 400, 100)))
myplot$plot

```
</div>
</div>

***

## ISS comparison
For this comparison, I used the file with adding Sweta or Avatar file add-up for Dx and compare to Raghu's ISS germline.
```{r}
# path <- fs::path("","Volumes","Gillis_Research","Christelle Colin-Leitzinger", "CHIP in Avatar")

Staging_ISS <- readxl::read_xlsx(paste0(path, "/Raghu MM/Other raw data/Staging_Germline_07272021.xlsx"),
                                 na = "NA") %>% 
  select("avatar_id", "collectiondt_germline", "Labs_Result_Date", "Final_Albumin", "Final_Beta2", "Final_LDH", "ISS") %>% 
  arrange(collectiondt_germline) %>% 
  distinct(avatar_id, .keep_all = TRUE) # remove the duplicate of patient 180
Diagnosis_ISS <- readxl::read_xlsx(paste0(path, "/Raghu MM/Other raw data/Staging_MM_Diagnosis_07262021_OUT.xlsx")) %>% 
  mutate(iss = str_replace(iss, pattern = "Unknown/Not Reported", replacement = NA_character_)) %>% 
  select(avatar_id, last_mrn = mrn, MM_date_dx = date_of_diagnosis, ISS_at_MMdx = iss)
```

### TRUE is when ISS at Dx is same as ISS at germline
```{r}
a <- left_join(germline_patient_data %>% select(avatar_id), Staging_ISS) %>% 
  full_join(., Diagnosis_ISS) %>% filter(!is.na(collectiondt_germline))
a <- germline_patient_data %>% select(ISS, ISS_at_MMdx)
a %>% 
  mutate(comparison = ISS == ISS_at_MMdx) %>% 
  group_by(comparison) %>% 
  summarise(sum = NROW(comparison)) %>% 
  ungroup() %>% 
  mutate(total = NROW(a)) %>% 
  mutate(percent = sum / NROW(a) * 100)
```

### ICC
```{r}
library(psych)

ICC_df <- a %>% select(ISS, ISS_at_MMdx) %>% 
  mutate(ISS = case_when(
    ISS == "I"    ~ 1,
    ISS == "II"    ~ 2,
    ISS == "III"    ~ 3,
  )) %>% 
  mutate(ISS_at_MMdx = case_when(
    ISS_at_MMdx == "I"    ~ 1,
    ISS_at_MMdx == "II"    ~ 2,
    ISS_at_MMdx == "III"    ~ 3,
  ))

ICC(ICC_df)
```

### Kappa

VALUE OF K	    STRENGTH OF AGREEMENT  
< 0	            Poor  
0.01 - 0.20	    Slight  
0.21-0.40	      Fair  
0.41-0.60	      Moderate  <--------- We are here :(  
0.61-0.80	      Substantial  
0.81 - 1.00	    Almost perfect  
```{r}
library(irr)

kappa2(ICC_df, "unweighted")
```





