---
title: "Disparities"
author: "Christelle Colin-Leitzinger"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: no
    theme: cerulean
    highlight: pygments
editor_options:
  chunk_output_type: console
---

<style type="text/css">
h1.title {
  font-size: 25px;
}
.figure {
   margin-top: 25px;
   margin-bottom: 25px;
}

table {
    margin-top: 25px;
    <!-- margin-bottom: 100px !important; -->
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.align='center'#, fig.show='hold'
                      )
options(gtsummary.print_engine = "gt")
options(gtsummary.as_gt.addl_cmds = "gt::tab_options(table.font.size = 12, data_row.padding = gt::px(1))")

```


<style>
div.blue { background-color:#0099CC; border-radius: 5px; padding: 20px; font-size: 38px}
</style>
<div class = "blue">

<span style="color: white;">Manuscript</span>

</div>
<br>

```{r library, include=FALSE}
library(tidyverse)
library(data.table)
library(VennDiagram)
library(viridis)
library(lubridate)
library(gtsummary)
library(survival)
library(survminer)
```

```{r load}
# Global_data <- readRDS("/Users/colinccm/Documents/GitHub/CHIP-Avatar/Global_data.rds")
germline_patient_data <- readRDS("/Users/colinccm/Documents/GitHub/CHIP-Avatar/germline_patient_data.rds")
path <- ("/Users/colinccm/Documents/GitHub/Gillis/no vpn data/CHIP in Avatar")

germline_patient_data <- germline_patient_data %>% 
  filter(is_patient_MM == "Yes" & is_MMDx_close_to_blood == "Yes") %>% 
  distinct(avatar_id, .keep_all = TRUE) %>% 
  mutate(CH_status = str_replace(CH_status, "No_CH", "No CH"))
```
<br>
<br>

***

# Introduction

The ORIEN Avatar^TM Research Program focuses on patients with advanced primary or metastatic disease, those with limited treatment options, as well as patients who are likely to develop progressive disease, and will capitalize on the disease-area expertise of ORIEN's research scientists.   
It generates massive amounts of genetic and clinical information on patients consenting to the Total Cancer CareÂ® (TCC) Protocol.   
ORIEN Avatar tracks patient clinical progressions over their lifetime to provide the richest data source for targeted cancer therapy advancement.  ORIEN Avatar data solution is optimized for oncology research by providing integrated longitudinal clinical data and whole exome, transcriptome and germline sequencing around consented oncology patients.   
We are focusing on patients with multiple myeloma disease.
We will focus on patient with active MM who had WES analyzed on their blood and CH were analyzed.

<br>

***

# I. MM Avatar data description

```{r}
germline_patient_data %>% 
  filter(Race == "Others" | Race == "Unknown" | Ethnicity == "Unknown") %>% 
  select(Race, Ethnicity)
```


<br>

From here, I removed the "Others" in raceeth.

```{r}
germline_patient_data <- germline_patient_data %>% 
  filter(raceeth != "Others") %>% 
  mutate(raceeth = factor(raceeth, levels=c("White Non-Hispanic", "Hispanic", "Black")))

```

<div class = "row">
<div class = "col-md-5">
```{r}
germ_data <- germline_patient_data %>% 
  mutate(Whole = "MM patients with WES") %>% 
  select(Age_at_MMonly_diagnosis, Gender, raceeth, Whole, ISS, ISS_grp) %>%
  tbl_summary(by = Whole, 
              sort = list(everything() ~ "frequency", ISS ~ "alphanumeric"),
              digits = list(c(Age_at_MMonly_diagnosis) ~ 2),
              missing = "no") %>% 
  bold_labels()
germ_data
```
</div>

<div class = "col-md-7">
```{r}
p <- qplot(x =Age_at_MMonly_diagnosis, data=subset(germline_patient_data,!is.na(Age_at_MMonly_diagnosis)), fill=..count.., geom="histogram") 
p + scale_fill_viridis_c(
  alpha = 1,
  begin = 0,
  end = 1,
  direction = 1,
  option = "D",
  values = NULL,
  space = "Lab",
  na.value = "grey50",
  guide = "colourbar",
  aesthetics = "fill"
) +
  theme_minimal(base_size = 16) +
  labs(x="Age at Diagnosis", y="Number of Patient", title="Repartition of Age at Diagnosis in MM Avatar")
```
</div>
</div>

<br>

Patients represented in MM Avatar data are characterized by a median age at diagnosis of `r inline_text(germ_data, variable = Age_at_MMonly_diagnosis, column = "MM patients with WES", pattern = "{median} years old")`. We saw slightly more male `r inline_text(germ_data, variable = Gender, level = "Male", column = "MM patients with WES")`, a majority of White Non-Hispanic `r inline_text(germ_data, variable = raceeth, level = "White Non-Hispanic", column = "MM patients with WES")`(Blacks `r inline_text(germ_data, variable = raceeth, level = "Black", column = "MM patients with WES")`, Hispanic `r inline_text(germ_data, variable = raceeth, level = "Hispanic", column = "MM patients with WES")`).
A higher proportion of patients have a lower clinical stage at blood with `r inline_text(germ_data, variable = ISS_grp, level = "I", column = "MM patients with WES")` patients with ISS I and `r inline_text(germ_data, variable = ISS_grp, level = "II-III", column = "MM patients with WES")` patients with ISS II and III combined.

<br>
<br>

***

# II. Age Distribution in Disparities

## Raceeth
```{r}
p <- ggplot(germline_patient_data, 
            aes(x=raceeth, y=Age_at_MMonly_diagnosis), fill=raceeth) + 
  geom_boxplot(color = c("darkred", "darkgreen", "blue")) + 
  theme_minimal(base_size = 23) +
  labs(x=NULL, y="Age at Diagnosis", title="Age of MM Diagnosis Distribution")
p + geom_jitter(shape=16, position=position_jitter(0.2)) +
  stat_compare_means()
```

```{r tbl for number}
raceeth_age <- germline_patient_data %>% 
  select(raceeth, Age_at_MMonly_diagnosis) %>%
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
```
<br>

Interestingly, we noticed a significant difference between disparities on the age at diagnosis. White Hispanics are diagnosed at a median age of `r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "White Non-Hispanic", pattern = "{median} years old")`. A significant earlier age at diagnosis is measured for Hispanics, `r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "Hispanic", pattern = "{median} years old")` , and for Blacks, `r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "Black", pattern = "{median} years old")`, compare to Whites Hispanics (`r inline_text(raceeth_age, variable = Age_at_MMonly_diagnosis, column = "p.value")`).

<br>
<br>

***

# III. RaceEthnicity

## Survivals

<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
raceeth_surv <- germline_patient_data

mysurv <- Surv(time = raceeth_surv$month_at_os, event = raceeth_surv$os_event)
myplot <- survfit(mysurv~raceeth, data = raceeth_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
ggsurvplot(myplot, data = raceeth_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           palette = c("blue", "darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")),
           # Censor
           censor = TRUE
) + guides(colour = guide_legend(nrow = 3))

tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Gender, ISS_grp, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_os, 
                             event = raceeth_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_os, 
             event = raceeth_surv$os_event) ~ raceeth + Age_at_MMonly_diagnosis + Gender + ISS_grp + HCT_ever, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
# eth_cox_os <- 
  tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
# eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
myplot <- survfit(mysurv~raceeth, data = raceeth_surv)
ggsurvplot(myplot, data = raceeth_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           palette = c("blue", "darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")
           ),
           # Censor
           censor = TRUE
) + guides(colour = guide_legend(nrow = 3))

tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Gender, ISS_grp, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_progression_drug, 
                             event = raceeth_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_progression_drug, 
             event = raceeth_surv$drug_progression_event) ~ raceeth + Age_at_MMonly_diagnosis + Gender + ISS_grp + HCT_ever, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
# eth_cox_pfs <- 
  tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
# eth_cox_pfs
```
</div>
</div>

<br>
<br>

## Delay to Treatment

```{r}
germline_patient_data %>% 
  select(time_to_drug, time_to_hct, raceeth) %>%
  tbl_summary(by = raceeth, 
              label = list(time_to_drug ~ "Time to drug (days)", time_to_hct ~ "Time to HCT (days)"),
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% bold_labels() %>% add_overall() %>% add_p() %>% bold_p(t = .05)
```

<br>
<br>

***

# IV. CH

```{r}
ch_char <- germline_patient_data %>% 
  mutate(Whole = "MM patients at time of blood with WES") %>% 
  select(Whole, CH_status) %>%
  tbl_summary(by = Whole, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels()
ch_char
```
<br>
<br>

Whole Exome Sequencing data identified CH mutations in `r inline_text(ch_char, variable = CH_status, level = "CH", column = "MM patients at time of blood with WES")`.
+ Add bioinfo results (oncoplot?).  
Type of genes affected...

<br>
<br>

```{r}
# ch_samples <- germline_patient_data %>%
#   select(CH_status) %>% 
#   tbl_summary(sort = list(everything() ~ "frequency")) %>% 
#   bold_labels()
# ch_samples

ch_demo <- germline_patient_data %>%
  select(Age_at_MMonly_diagnosis, 
         Gender, raceeth, ISS, ISS_grp,
         CH_status) %>% 
  tbl_summary(by = CH_status, 
              label = list(Age_at_MMonly_diagnosis ~ "Age at MM diagnosis", CH_status ~ "CH status"),
              sort = list(everything() ~ "frequency", ISS ~ "alphanumeric", ISS_grp ~ "alphanumeric"),
              digits = list(c(Age_at_MMonly_diagnosis) ~ 2),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
ch_demo
```
<br>

A significant difference in the age at diagnosis was observed between CH and No CH patients with an older age for CH patients (`r inline_text(ch_demo, variable = Age_at_MMonly_diagnosis, column = "CH")` vs `r inline_text(ch_demo, variable = Age_at_MMonly_diagnosis, column = "No CH")`, `r inline_text(ch_demo, variable = Age_at_MMonly_diagnosis, column = "p.value")`). While Gender and race/ethnicity are similarly distributed, we noticed a higher proportion of patients with more severe clinical stage (II-III) for CH patients (`r inline_text(ch_demo, variable = ISS_grp, column = "p.value")`).

<br>
<br>

***

# IV. CH & RaceEthnicity
### PFS
```{r, fig.height = 7}
mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
ggsurvplot(myplot, data = raceeth_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           color = "raceeth",
           linetype = "CH_status",
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 3,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")
           ),
           # Censor
           censor = TRUE
)+ guides(colour = guide_legend(nrow = 3), linetype = guide_legend(nrow = 2))
```

```{r}
tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Gender, ISS_grp, HCT_ever, CH_status) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_progression_drug, 
                             event = raceeth_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_progression_drug, 
             event = raceeth_surv$drug_progression_event) ~ raceeth + Age_at_MMonly_diagnosis + Gender + ISS_grp + HCT_ever + CH_status, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
```

### OS
```{r, fig.height = 7}
mysurv <- Surv(time = raceeth_surv$month_at_os, event = raceeth_surv$os_event)
myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
ggsurvplot(myplot, data = raceeth_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           color = "raceeth",
           linetype = "CH_status",
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 3,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")
           ),
           # Censor
           censor = TRUE
)+ guides(colour = guide_legend(nrow = 3), linetype = guide_legend(nrow = 2))
```

```{r}
tbl1 <- raceeth_surv %>% select(raceeth, Age_at_MMonly_diagnosis, Gender, ISS_grp, HCT_ever, CH_status) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = raceeth_surv$month_at_os, 
                             event = raceeth_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = raceeth_surv$month_at_os, 
                   event = raceeth_surv$os_event) ~ raceeth + Age_at_MMonly_diagnosis + Gender + ISS_grp + HCT_ever + CH_status, data =  raceeth_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
```
<br>
<br>


***
```{r, fig.height = 7}
# mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
# myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
# ggsurvplot(myplot, data = raceeth_surv, facet.by = "MM_age_cat5",
#            title = "PFS",
#            font.main = c(24, "bold", "black"),
#            font.x = c(20, "bold", "black"),
#            font.y = c(20, "bold", "black"),
#            font.legend = c(20, "bold", "black"),
#            font.tickslab = c(18, "bold", "black"),
#            size = 1.5,
# color = "raceeth",
#            linetype = "CH_status",
#            xlab = "Time in months"
# )+ guides(colour = guide_legend(nrow = 4))
```

```{r, fig.height = 7}
# mysurv <- Surv(time = raceeth_surv$month_at_progression_drug, event = raceeth_surv$drug_progression_event)
# myplot <- survfit(mysurv~raceeth + CH_status, data = raceeth_surv)
# ggsurvplot(myplot, data = raceeth_surv, facet.by = "MM_age_cat3",
#            title = "PFS",
#            font.main = c(24, "bold", "black"),
#            font.x = c(20, "bold", "black"),
#            font.y = c(20, "bold", "black"),
#            font.legend = c(20, "bold", "black"),
#            font.tickslab = c(18, "bold", "black"),
#            size = 1.5,
#            
#            xlab = "Time in months"
# )+ guides(colour = guide_legend(nrow = 4))
```
<br>
<br>

## WNH
<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
nhis_surv <- germline_patient_data %>% 
  filter(raceeth == "White Non-Hispanic")

mysurv <- Surv(time = nhis_surv$month_at_os, event = nhis_surv$os_event)
myplot <- survfit(mysurv~CH_status, data = nhis_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
ggsurvplot(myplot, data = nhis_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "Ethnicity",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           # palette = c("darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")),
           # Censor
           censor = TRUE
)

tbl1 <- nhis_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Gender, ISS_grp, Drugs_ever, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = nhis_surv$month_at_os, 
                             event = nhis_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = nhis_surv$month_at_os, 
             event = nhis_surv$os_event) ~ CH_status + Age_at_MMonly_diagnosis + Gender + ISS_grp + Drugs_ever + HCT_ever, data =  nhis_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_os <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = nhis_surv$month_at_progression_drug, event = nhis_surv$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = nhis_surv)
ggsurvplot(myplot, data = nhis_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "Ethnicity",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           palette = c("darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")
           ),
           # Censor
           censor = TRUE
)

tbl1 <- nhis_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Gender, ISS_grp, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = nhis_surv$month_at_progression_drug, 
                             event = nhis_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = nhis_surv$month_at_progression_drug, 
             event = nhis_surv$drug_progression_event) ~ CH_status + Age_at_MMonly_diagnosis + Gender + ISS_grp + HCT_ever, data =  nhis_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_pfs <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_pfs
```
</div>
</div>

<br>
<br>

## Bl
<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
bl_surv <- germline_patient_data %>% 
  filter(raceeth == "Black")

mysurv <- Surv(time = bl_surv$month_at_os, event = bl_surv$os_event)
myplot <- survfit(mysurv~CH_status, data = bl_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
ggsurvplot(myplot, data = bl_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "Ethnicity",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           # palette = c("darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")),
           # Censor
           censor = TRUE
)

tbl1 <- bl_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Gender, ISS_grp, Drugs_ever, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = bl_surv$month_at_os, 
                             event = bl_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = bl_surv$month_at_os, 
             event = bl_surv$os_event) ~ CH_status + Age_at_MMonly_diagnosis + Gender + ISS_grp + Drugs_ever + HCT_ever, data =  bl_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_os <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = bl_surv$month_at_progression_drug, event = bl_surv$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = bl_surv)
ggsurvplot(myplot, data = bl_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "Ethnicity",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           palette = c("darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")
           ),
           # Censor
           censor = TRUE
)

tbl1 <- bl_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Gender, ISS_grp, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = bl_surv$month_at_progression_drug, 
                             event = bl_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = bl_surv$month_at_progression_drug, 
             event = bl_surv$drug_progression_event) ~ CH_status + Age_at_MMonly_diagnosis + Gender + ISS_grp + HCT_ever, data =  bl_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_pfs <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_pfs
```
</div>
</div>

<br>
<br>

## H

<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
his_surv <- germline_patient_data %>% 
  filter(raceeth == "Hispanic")

mysurv <- Surv(time = his_surv$month_at_os, event = his_surv$os_event)
myplot <- survfit(mysurv~CH_status, data = his_surv)
# jpeg(paste0(path, "/Figures/Survivals/Demographic/OS Ethnicity.jpeg"), width = 1200, height = 900)
ggsurvplot(myplot, data = his_surv,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "Ethnicity",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           # palette = c("darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")),
           # Censor
           censor = TRUE
)

tbl1 <- his_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Gender, ISS_grp, Drugs_ever, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = his_surv$month_at_os, 
                             event = his_surv$os_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = his_surv$month_at_os, 
             event = his_surv$os_event) ~ CH_status + Age_at_MMonly_diagnosis + Gender + ISS_grp + Drugs_ever + HCT_ever, data =  his_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_os <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_os
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = his_surv$month_at_progression_drug, event = his_surv$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = his_surv)
ggsurvplot(myplot, data = his_surv,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "Ethnicity",
           # # legend.labs = c("Hispanic", "Non-Hispanic"),
           palette = c("darkred", "darkgreen"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.3,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black")
           ),
           # Censor
           censor = TRUE
)

tbl1 <- his_surv %>% select(CH_status, Age_at_MMonly_diagnosis, Gender, ISS_grp, HCT_ever) %>% 
  tbl_uvregression(method = survival::coxph, 
                   y = (Surv(time = his_surv$month_at_progression_drug, 
                             event = his_surv$drug_progression_event)),
                   exponentiate = TRUE) %>% bold_p(t = .05) %>% add_nevent() %>% 
  bold_labels() %>% italicize_levels()
tbl2 <- coxph(Surv(time = his_surv$month_at_progression_drug, 
             event = his_surv$drug_progression_event) ~ CH_status + Age_at_MMonly_diagnosis + Gender + ISS_grp + HCT_ever, data =  his_surv) %>%
  tbl_regression(exponentiate = TRUE) %>% bold_p(t = .05)
eth_cox_pfs <- tbl_merge(list(tbl1, tbl2), tab_spanner = c("**Univariable**", "**Multivariable**"))
eth_cox_pfs
```
</div>
</div>

<br>

<!-- Age is significantly associated with OS `r inline_text(tbl1, variable = Age_at_MMonly_diagnosis, pattern = "(OR {estimate}; 95% CI {conf.low}, {conf.high}; {p.value})")`. -->

<br>

<!-- I thought it could be a year effect compare to Taiga said (maybe earlier it wasn't a difference like he said but now that Hispanic may have more access to care we can see the difference) but no. -->

```{r}
# mysurv <- Surv(time = ethnicity_surv$month_at_progression_drug, event = ethnicity_surv$drug_progression_event)
# myplot <- survfit(mysurv~Ethnicity1+diagnosis_MM_year, data = ethnicity_surv)
# ggsurvplot_facet(myplot, data = ethnicity_surv, facet.by = "diagnosis_MM_year",
#            title = "PFS",
#            font.main = c(24, "bold", "black"),
#            font.x = c(20, "bold", "black"),
#            font.y = c(20, "bold", "black"),
#            font.legend = c(10, "bold", "black"),
#            font.tickslab = c(18, "bold", "black"),
#            size = 1.5,
#            
#            xlab = "Time in months",
#            palette = c("darkred", "darkgreen"),
#            conf.int = FALSE
# )
```t

***

# Drugs (only regimen1)

VCD?
```{r}
germline_patient_data %>% 
  mutate(Whole = "MM patients with WES") %>% 
  select(regimen_category_1, raceeth) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
```
<br>

More than 20 patients
```{r}
germline_patient_data %>% 
  group_by(first_regimen_name) %>% 
  mutate(n = n()) %>% 
  ungroup() %>% 
  filter(n > 20) %>% 
  # select(first_regimen_name, n)
  mutate(Whole = "MM patients with WES") %>% 
  select(first_regimen_name, raceeth) %>% 
  tbl_summary(by = raceeth, 
              sort = list(everything() ~ "frequency"),
              missing = "no") %>% 
  bold_labels() %>% add_p() %>% bold_p(t = .05)
```
<br>

***

# Not shown

## CH

### Survivals

<div class = "row">
<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = germline_patient_data$month_at_os, event = germline_patient_data$os_event)
myplot <- survfit(mysurv~CH_status, data = germline_patient_data)
# jpeg(paste0(path, "/Figures/Survivals/CHIP/OS by CH.jpeg"), width = 1200, height = 900)
ggsurvplot(myplot, data = germline_patient_data,
           title = "OS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "", 
           # # legend.labs = c("CH", "No CH"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.17,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black"))
)
```
</div>

<div class = "col-md-6">
```{r, fig.height = 7}
mysurv <- Surv(time = germline_patient_data$month_at_progression_drug, event = germline_patient_data$drug_progression_event)
myplot <- survfit(mysurv~CH_status, data = germline_patient_data)
ggsurvplot(myplot, data = germline_patient_data,
           title = "PFS",
           font.main = c(24, "bold", "black"),
           font.x = c(20, "bold", "black"),
           font.y = c(20, "bold", "black"),
           font.legend = c(20, "bold", "black"),
           font.tickslab = c(18, "bold", "black"),
           size = 1.5,
           
           xlab = "Time in months", 
           legend = "top",
           legend.title = "", 
           # # legend.labs = c("CH", "No CH"),
           pval = TRUE,
           conf.int = FALSE,
           # Add risk table
           tables.height = 0.17,
           risk.table.title = "Risk table (number(%))",
           risk.table = "abs_pct",
           risk.table.y.text = FALSE,
           risk.table.fontsize = 4,
           tables.theme = theme_survminer(base_size = 5,
                                          font.main = c(16, "bold", "black"),
                                          font.x = c(16, "bold", "black"),
                                          font.y = c(16, "bold", "transparent"),
                                          font.tickslab = c(19, "bold", "black"))
)
```
</div>
</div>

***
